### ISSUE TITLE
Fix processing of partial JSON payloads from LMS

### ISSUE THREAD

**Issue by kaaholst** [kaaholst/android-squeezer#645]
Fixes: #643

Signed-off-by: Gregory Moyer <moyerg@syphr.com>

---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/nikclayton/android-squeezer/644)

Original issue reported on github.com/nikclayton/android-squeezer by syphr42 on 2020-06-07T16:41:40Z

**Comment by kaaholst** [comment]
Well spotted, thanks for the contribution.

One question:
How do I trigger/reproduce this?

Original issue reported on github.com/nikclayton/android-squeezer by kaaholst on 2020-06-09T16:19:22Z

**Comment by kaaholst** [comment]
It seems to happen reliably when I added more players. I'm assuming that any sufficiently long messages from the server could get broken up this way. However, the messages sent from LMS did not appear to always be broken in the same spot which made it a little tricky to track down.

Original issue reported on github.com/nikclayton/android-squeezer by syphr42 on 2020-06-10T21:56:11Z

**Comment by kaaholst** [comment]
Hmm ... I haven't been able to reproduce this. I tried up to 4 players, different LMS versions and on different os. After the serverstatus and playerstatus messages we order the home menu which is the longest response of about 12k, but even that comes in one line.
I would have liked to know what's causing this. It's a good reliability fix though, so I merged it in.

Thanks for the reply, and thanks again for the contribution. 

Original issue reported on github.com/nikclayton/android-squeezer by kaaholst on 2020-06-11T19:43:50Z
